libsolidity/ast/AST.cpp

```diff
@@ -247,6 +247,21 @@ FunctionDefinition const* ContractDefinition::nextConstructor(ContractDefinition
 	return nullptr;
 }
 
+bool ContractDefinition::dependenciesAreCyclic(
+	std::set<ContractDefinition const*> const& _seenContracts
```

Comment:
Can you remove this parameter and use the breadth first search utility funciton instead?

Reply:
Agree

Reply:
Turns out we have an actual CycleDetector class which I used in place now.